components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/JDBCConfigDAOImpl.java

```diff
@@ -0,0 +1,139 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.entitlement.dao;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.database.utils.jdbc.NamedPreparedStatement;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.entitlement.EntitlementException;
+import org.wso2.carbon.identity.entitlement.PDPConstants;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import static org.wso2.carbon.identity.entitlement.PDPConstants.Algorithms.DENY_OVERRIDES;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.EntitlementTableColumns.CONFIG_KEY;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.EntitlementTableColumns.CONFIG_VALUE;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.EntitlementTableColumns.TENANT_ID;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.CREATE_POLICY_COMBINING_ALGORITHM_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.GET_POLICY_COMBINING_ALGORITHM_SQL;
+import static org.wso2.carbon.identity.entitlement.dao.DAOConstants.SQLQueries.UPDATE_POLICY_COMBINING_ALGORITHM_SQL;
+
+public class JDBCConfigDAOImpl implements ConfigDAO {
+
+    private static final Log LOG = LogFactory.getLog(JDBCConfigDAOImpl.class);
+
+    /**
+     * Gets the policy combining algorithm name of the PDP.
+     *
+     * @return policy combining algorithm name.
+     */
+    @Override
+    public String getGlobalPolicyAlgorithmName() {
+
+        String algorithm = getPolicyCombiningAlgorithm();
+        if (StringUtils.isBlank(algorithm)) {
+            algorithm = DENY_OVERRIDES;
+        }
+
+        return algorithm;
+    }
+
+    /**
+     * Persists the policy combining algorithm into the data store.
+     *
+     * @param policyCombiningAlgorithm policy combining algorithm name to persist.
+     * @throws EntitlementException throws if fails.
+     */
+    @Override
+    public void setGlobalPolicyAlgorithm(String policyCombiningAlgorithm) throws EntitlementException {
+
+        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+
+        // Check the existence of the algorithm
+        String algorithm = getPolicyCombiningAlgorithm();
+        if (StringUtils.isBlank(algorithm)) {
+            setPolicyCombiningAlgorithm(policyCombiningAlgorithm, tenantId);
+        } else {
+            updatePolicyCombiningAlgorithm(policyCombiningAlgorithm, tenantId);
+        }
+    }
+
+    public String getPolicyCombiningAlgorithm() {
```

Comment:
`getPolicyCombiningAlgorithm()`, `setPolicyCombiningAlgorithm()` & `updatePolicyCombiningAlgorithm()` are the real DAO methods we have.. There are no business logics in those methods, and only handle storage logic.

Shall we add method comments with some common pattern to quickly identify what are the pure DAO methods we have?